### PR TITLE
Clang tidy Part 1: Ui Project Function Names

### PR DESCRIPTION
**Pull request by duncanspumpkin**
```
Checks: 'readability-identifier-naming'
CheckOptions:
  - key: readability-identifier-naming.FunctionCase 
    value: 'CamelCase
```

### CODE REVIEW

File: src/openrct2-ui/WindowManager.cpp [hunk 1]
```diff
@@ -32,51 +32,51 @@ class WindowManager final : public IWindowManager
     void Init() override
     {
         ThemeManagerInitialise();
-        window_new_ride_init_vars();
+        window_staff_list_init_vars();
```
Comment (duncanspumpkin) [inline src/openrct2-ui/WindowManager.cpp:35]: incorrect substitution.

File: src/openrct2-ui/drawing/engines/opengl/DrawCommands.h [hunk 2]
```diff
@@ -27,35 +27,37 @@ template<typename T> class CommandBatch
         : _numInstances(0)
     {
     }
-    [[nodiscard]] bool empty() const
+
+    // NOLINTBEGIN
```
Comment (Broxzier) [inline src/openrct2-ui/drawing/engines/opengl/DrawCommands.h:31]: Only ignore the `readability-identifier-naming` check, not all:
```suggestion
    // NOLINTBEGIN(readability-identifier-naming)
```

File: src/openrct2-ui/drawing/engines/opengl/DrawCommands.h [hunk 3]
```diff
@@ -27,35 +27,37 @@ template<typename T> class CommandBatch
...
+    [[nodiscard]] bool empty() const // NOLINT
```
Comment (Broxzier) [inline src/openrct2-ui/drawing/engines/opengl/DrawCommands.h:32]: Are these comments necessary, while they are already inside a `NOLINTBEGIN`/`NOLINTEND` block?

File: src/openrct2-ui/drawing/engines/opengl/OpenGLAPI.h [hunk 4]
```diff
@@ -11,6 +11,7 @@
 
 #include <openrct2/common.h>
 
+// NOLINTBEGIN
```
Comment (Broxzier) [inline src/openrct2-ui/drawing/engines/opengl/OpenGLAPI.h:14]:
```suggestion
// NOLINTBEGIN(readability-identifier-naming)
```

File: src/openrct2-ui/drawing/engines/opengl/OpenGLAPI.h [hunk 5]
```diff
@@ -112,6 +113,7 @@ using PFNGLGETTEXIMAGEPROC = void(APIENTRYP)(GLenum target, GLint level, GLenum
 #    undef OPENGL_PROC
 
 #endif /* OPENGL_NO_LINK */
+// NOLINTEND
```
Comment (Broxzier) [inline src/openrct2-ui/drawing/engines/opengl/OpenGLAPI.h:116]:
```suggestion
// NOLINTEND(readability-identifier-naming)
```

File: src/openrct2-ui/drawing/engines/opengl/DrawCommands.h [hunk 6]
```diff
@@ -64,30 +66,31 @@ template<typename T> class CommandBatch
...
+    typename std::vector<T>::const_iterator end() const // NOLINT
     {
         return _instances.cbegin() + _numInstances;
     }
-    typename std::vector<T>::const_iterator cend() const
+    typename std::vector<T>::const_iterator cend() const // NOLINT
     {
         return _instances.cbegin() + _numInstances;
     }
+    // NOLINTEND
```
Comment (Broxzier) [inline src/openrct2-ui/drawing/engines/opengl/DrawCommands.h:93]:
```suggestion
    // NOLINTEND(readability-identifier-naming)
```

File: src/openrct2-ui/drawing/engines/opengl/OpenGLAPIProc.h [hunk 7]
```diff
@@ -11,6 +11,7 @@
 #    error "Do not include OpenGLAPIProc.h directly. Include OpenGLAPI.h instead."
 #endif
 
+// NOLINTBEGIN
```
Comment (Broxzier) [inline src/openrct2-ui/drawing/engines/opengl/OpenGLAPIProc.h:14]:
```suggestion
// NOLINTBEGIN(readability-identifier-naming)
```

File: src/openrct2-ui/drawing/engines/opengl/OpenGLAPIProc.h [hunk 8]
```diff
@@ -84,3 +85,4 @@ OPENGL_PROC(PFNGLVERTEXATTRIBPOINTERPROC, glVertexAttribPointer)
 OPENGL_PROC(PFNGLDRAWARRAYSINSTANCEDPROC, glDrawArraysInstanced)
 OPENGL_PROC(PFNGLVERTEXATTRIBDIVISORPROC, glVertexAttribDivisor)
 OPENGL_PROC(PFNGLBLENDFUNCSEPARATEPROC, glBlendFuncSeparate)
+// NOLINTEND
```
Comment (Broxzier) [inline src/openrct2-ui/drawing/engines/opengl/OpenGLAPIProc.h:88]:
```suggestion
// NOLINTEND(readability-identifier-naming)
```

File: src/openrct2-ui/drawing/engines/opengl/DrawCommands.h (hunk 2 above)
Comment (Broxzier) [reply src/openrct2-ui/drawing/engines/opengl/DrawCommands.h:31]: Not sure if this already exists in the version we use. This seems to be a relatively new feature.

Comment (duncanspumpkin) [reply src/openrct2-ui/drawing/engines/opengl/DrawCommands.h:31]: Yeah it didn't work for me so i had to add the NOLINT comments as well

File: src/openrct2-ui/WindowManager.cpp (hunk 1 above)
Comment (ZehMatt) [reply src/openrct2-ui/WindowManager.cpp:35]: How did clang-tidy manage to do that?

Comment (duncanspumpkin) [reply src/openrct2-ui/WindowManager.cpp:35]: No idea but its very annoying that it does this.

File: src/openrct2-ui/windows/ClearScenery.cpp [hunk 9]
```diff
@@ -81,7 +81,7 @@ class CleanSceneryWindow final : public Window
 
     void OnClose() override
     {
-        if (clear_scenery_tool_is_active())
+        if (scenery_tool_is_active())
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/ClearScenery.cpp:84]: mistaken substitution

File: src/openrct2-ui/windows/ClearScenery.cpp [hunk 10]
```diff
@@ -155,7 +155,7 @@ class CleanSceneryWindow final : public Window
     {
         frame_no++;
         // Close window if another tool is open
-        if (!clear_scenery_tool_is_active())
+        if (!scenery_tool_is_active())
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/ClearScenery.cpp:158]: mistaken substitution

File: src/openrct2-ui/windows/Land.cpp [hunk 11]
```diff
@@ -89,7 +89,7 @@ class LandWindow final : public Window
     void OnClose() override
     {
         // If the tool wasn't changed, turn tool off
-        if (land_tool_is_active())
+        if (scenery_tool_is_active())
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/Land.cpp:92]: mistake

File: src/openrct2-ui/windows/Land.cpp [hunk 12]
```diff
@@ -209,7 +209,7 @@ class LandWindow final : public Window
 
     void OnUpdate() override
     {
-        if (!land_tool_is_active())
+        if (!scenery_tool_is_active())
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/Land.cpp:212]: mistake

File: src/openrct2-ui/windows/Water.cpp [hunk 13]
```diff
@@ -58,7 +58,7 @@ class WaterWindow final : public Window
     void OnClose() override
     {
         // If the tool wasn't changed, turn tool off
-        if (water_tool_is_active())
+        if (scenery_tool_is_active())
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/Water.cpp:61]: mistake

File: src/openrct2-ui/windows/Water.cpp [hunk 14]
```diff
@@ -101,7 +101,7 @@ class WaterWindow final : public Window
     void OnUpdate() override
     {
         // Close window if another tool is open
-        if (!water_tool_is_active())
+        if (!scenery_tool_is_active())
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/Water.cpp:104]: mistake

File: src/openrct2-ui/windows/About.cpp [hunk 15]
```diff
@@ -91,25 +91,25 @@ static uint64_t window_about_page_enabled_widgets[] = {
     DEFAULT_ENABLED_WIDGETS,
 };
 
-static void window_about_openrct2_mouseup(rct_window *w, rct_widgetindex widgetIndex);
-static void window_about_openrct2_paint(rct_window *w, rct_drawpixelinfo *dpi);
-static void window_about_openrct2_invalidate(rct_window *w);
+static void WindowAboutOpenrct2Mouseup(rct_window *w, rct_widgetindex widgetIndex);
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/About.cpp:94]:
```suggestion
static void WindowAboutOpenRCT2Mouseup(rct_window *w, rct_widgetindex widgetIndex);
```
Probably best to do a (case sensitive) find-and-replace.

File: src/openrct2-ui/windows/About.cpp [hunk 16]
```diff
@@ -91,25 +91,25 @@ static uint64_t window_about_page_enabled_widgets[] = {
...
+static void WindowAboutOpenrct2Paint(rct_window *w, rct_drawpixelinfo *dpi);
+static void WindowAboutOpenrct2Invalidate(rct_window *w);
 
-static void window_about_rct2_mouseup(rct_window *w, rct_widgetindex widgetIndex);
-static void window_about_rct2_paint(rct_window *w, rct_drawpixelinfo *dpi);
-static void window_about_openrct2_common_paint(rct_window *w, rct_drawpixelinfo *dpi);
+static void WindowAboutRct2Mouseup(rct_window *w, rct_widgetindex widgetIndex);
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/About.cpp:98]:
```suggestion
static void WindowAboutRCT2Mouseup(rct_window *w, rct_widgetindex widgetIndex);
```

See previous statement.

File: src/openrct2-ui/windows/Map.cpp [hunk 17]
```diff
@@ -589,7 +589,7 @@ static void window_map_scrollmousedown(rct_window* w, int32_t scrollIndex, const
         window_scroll_to_location(mainWindow, { mapCoords, mapZ });
     }
 
-    if (land_tool_is_active())
+    if (scenery_tool_is_active())
```
Comment (Broxzier) [inline src/openrct2-ui/windows/Map.cpp:592]: Wrong function

Comment (duncanspumpkin) [reply src/openrct2-ui/windows/Map.cpp:592]: Argh thought i caught all of these

Comment (Broxzier) [reply src/openrct2-ui/windows/Map.cpp:592]: I'm pretty sure this is the last one. I went over this diff carefully, with the help of regex to find this only exception.

**Pull request merged**